Sources/Deferred.swift

```diff
@@ -28,9 +28,15 @@ private struct DispatchBlockMarker: CallbacksList {
         // Executing the block "unblocks" it, calling all the `_notify` blocks
         block()
     }
-    
-    func notify(upon queue: dispatch_queue_t, body: dispatch_block_t) {
-        dispatch_block_notify(block, queue, body)
+
+    func notify(upon executor: ExecutorType, body: dispatch_block_t) {
```

Comment:
Hmm, very good point. Do you "notify **with** an executor", "notify an executor", "notify with an executor"?